autobisect/bisect.py

```diff
@@ -181,6 +181,82 @@ def __init__(
 
         self.build_manager = BuildManager(config)
 
+    def _get_daily_builds(self):
+        """
+        Create build range containing one build per day
+        """
+        start = self.start.datetime + timedelta(days=1)
+        end = self.end.datetime - timedelta(days=1)
+        LOG.info(f"Enumerating daily builds: {start} - {end}")
+
+        return BuildRange.new(start, end)
+
+    def _get_pushdate_builds(self):
+        """
+        Create build range containing all builds per pushdate
+        """
+        start = self.start.datetime
+        end = self.end.datetime
+        LOG.info(f"Enumerating pushdate builds: {start} - {end}")
+
+        builds = []
+        for dt in [start, end]:
+            date = dt.strftime("%Y-%m-%d")
+            for build in Fetcher.iterall(self.target, self.branch, date, self.flags):
+                # Only keep builds after the start and before the end boundaries
+                if self.end.datetime > build.datetime > self.start.datetime:
```

Comment:
This check seems redundant. If we are iterating over datetimes between start and end then shouldn't this always be true?

Reply:
No.  The reason being is that Fetcher.iterall() will only accept a date without time information.  Those builds will be on the same day of start and end but with their timestamp, may fall outside of that range.

Reply:
OK, next question should at least one of these be inclusive (line 207)? You might miss a build if hit a build date right on. Very unlikely but possible I think.

Reply:
No.  At this point, the builds at self.start and self.end have already been evaluated.  We only want to test builds within those bounds.